@xen-orchestra/backups/_cleanVm.js

```diff
@@ -106,12 +106,17 @@ const listVhds = async (handler, vmDir) => {
             filter: file => isVhdFile(file) || INTERRUPTED_VHDS_REG.test(file),
           })
           aliases[vdiDir] = list.filter(vhd => isVhdAlias(vhd)).map(file => `${vdiDir}/${file}`)
-          list.forEach(file => {
+
+          await asyncMap(list, async file => {
             const res = INTERRUPTED_VHDS_REG.exec(file)
             if (res === null) {
               vhds.add(`${vdiDir}/${file}`)
             } else {
-              interruptedVhds.set(`${vdiDir}/${res[1]}`, `${vdiDir}/${file}`)
+              const mergeState = JSON.parse(await handler.readFile(file))
+              interruptedVhds.set(`${vdiDir}/${res[1]}`, {
+                statePath: `${vdiDir}/${file}`,
+                childPath: mergeState.childPath,
```

Comment:
What about older merge state without this entry?

Reply:
it fall back to parent/child : https://github.com/vatesfr/xen-orchestra/pull/6349/files#diff-d093af5f386c81f2203a927f2578e5d2c99d78a3d1d3dae0681036cdab9529b1R440

Reply:
Perfect :+1: